scripts/build/gen_relocate_app.py

```diff
@@ -53,7 +53,7 @@
 """
 
 SECTION_LOAD_MEMORY_SEQ = """
-        __{0}_{1}_rom_start = LOADADDR(_{2}_{3}_SECTION_NAME);
+        __{0}_{1}_rom_start = LOADADDR(.{0}_{1}_reloc);
```

Comment:
just note that not very many sections uses `.` in their section names.

Looking at it, it seems some of linker code has bitrotten over the years, cause I notice some sections are having names like: `_SRAM_DATA_SECTION_NAME`

Reply:
ahh, seems those were coming from the relocation script and this PR actually fixes those.

Thanks for that 👍 

Question which remains is thus, should we have a leading `.` or not ?
It's not really consistent in here either:
https://github.com/zephyrproject-rtos/zephyr/blob/e40254a44ad3eaeb294e260a9fee324c5d129746/include/zephyr/linker/sections.h#L18-L20
vs.
https://github.com/zephyrproject-rtos/zephyr/blob/e40254a44ad3eaeb294e260a9fee324c5d129746/include/zephyr/linker/sections.h#L47-L49
(note, there are a lot more of those in same file)

Reply:
Yes, that is one of the names this bit of the patch changes. `_SRAM_DATA_SECTION_NAME` becomes the slightly less surprising `.sram_data_reloc`.
While it's true most sections don't have a . prefix, the memory region specific ones do (.sram_noinit, .dtcm_bss etc). I'm just trying to follow the convention established elsewhere in Zephyr. I'm open for better suggestions.

Reply:
>  I'm just trying to follow the convention established elsewhere in Zephyr. 

that's my problem. There is not a clear convention on this.

Looking at ELF spec, section `Special Sections`. p. 1-16, it states:
> Section names with a dot (.) prefix are reserved for the system, although applications may use
these sections if their existing meanings are satisfactory. Applications may use names without
the prefix to avoid conflicts with system sections. 

under normal circumstances I would probably avoid the `.` to remove any collision risks.
But your `_reloc` effectively ensures the same, thus I see no problems with the leading `.`.

So I guess i'm fine with either approach, but would like to know if others are having opinion on this.

@carlocaione do you have any preferences on this ?

Reply:
Given that we are dealing with kernel code I would follow the ELF spec and just prefix all the kernel sections with `.` and leaving the `.` out only for the `app_*` sections.

Reply:
@carlocaione  should that be understood so that you actually propose to adjust https://github.com/zephyrproject-rtos/zephyr/blob/e40254a44ad3eaeb294e260a9fee324c5d129746/include/zephyr/linker/sections.h to follow that as a rule ?

Reply:
> @carlocaione should that be understood so that you actually propose to adjust https://github.com/zephyrproject-rtos/zephyr/blob/e40254a44ad3eaeb294e260a9fee324c5d129746/include/zephyr/linker/sections.h to follow that as a rule ?

Yes, hopefully it won't be a huge work?